### DISCUSSION TITLE
Allow to enable/disable queue generation globally in config

### DISCUSSION THREAD

**Discussion by Okipa** [spatie/laravel-medialibrary#2088] [Ideas] [answered]
In basic apps with standard page reload, I always have to add the `->nonQueued()` argument for every single conversion in order to avoid users to get no generated images after they upload a new one (they would have to wait a bit and reload the page one more time to get the image finally displayed: this is not an ergonomically desirable behavior).

Once this has been setup with `->nonQueued()`, all images are generated synchronously, even in local, which can be VERY annoying when we have to migrate and seed everything 🤯.

I was thinking about a way to easily allow to enable/disable queuing for conversions generation, that could be done with a simple and non-breaking config file line addition. For example:
```php
/*
 * Set the default conversions queued generation behavior.
 * Set `true` to have your conversions generated in queue by default and `false` if you prefer to get them generated synchronously.
 */
'conversions_generated_in_queue' => env('CONVERSIONS_GENERATED_IN_QUEUE', true),
```

Would you accept a PR taking care of this, with related tests and documentation addition ?

**Comment by freekmurze** [comment, marked as answer]
I like the proposed functionality. Name the config key `queue_conversions_by_default`. Also add a method `queue()` to queue a conversion when `conversions_generated_in_queue` is set to false.

> with related tests and documentation addition
👍

**Reply by Okipa** [reply]
Here it is! https://github.com/spatie/laravel-medialibrary/pull/2094
